src/backends/beamcoder.ts

```diff
@@ -390,19 +391,22 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
         // Read packets until we have a frame which is closest to targetPTS
         while ((this.#packet || this.#frames.length !== 0) && closestFramePTS < targetPTS) {
             VERBOSE && console.log('packet si:', this.#packet?.stream_index, 'pts:', this.#packet?.pts, 'frames:', this.#frames?.length);
-            VERBOSE && console.log('frames', this.#frames?.length, 'frames.pts:', this.#frames?.map(f => f.pts), '-> target.pts:', targetPTS);
+            VERBOSE && console.log('frames', this.#frames?.length, 'frames.pts:', JSON.stringify(this.#frames?.map(f => f.pts)), '-> target.pts:', targetPTS);
 
             // packet contains frames
             if (this.#frames.length !== 0) {
                 // filter the frames
                 const filteredResult = await this.#filterer.filter([{ name: 'in0:v', frames: this.#frames }]);
                 filteredFrames = filteredResult.flatMap(r => r.frames);
-                VERBOSE && console.log('filteredFrames', filteredFrames.length, 'filteredFrames.pts:', filteredFrames.map(f => f.pts), '-> target.pts:', targetPTS);
+                VERBOSE && console.log('filteredFrames', filteredFrames.length, 'filteredFrames.pts:', JSON.stringify(filteredFrames.map(f => f.pts)), '-> target.pts:', targetPTS);
 
                 // get the closest frame to our target presentation timestamp (PTS)
                 // Beamcoder returns decoded packet frames as follows: [1000, 2000, 3000, 4000]
+                // If we're looking for a frame at 0, we want to return the frame at 1000
                 // If we're looking for a frame at 2500, we want to return the frame at 2000
-                const closestFrame = filteredFrames.reverse().find(f => f.pts <= targetPTS);
+                const closestFrame = (this.#packetReadCount === 1 && filteredFrames[0].pts > targetPTS)
```

Comment:
will this work with looping too? like if we seek back to an earlier timestamp. I'm not sure how framefusion handles that but could be a case to consider? So if we play and then loop back to the start (maybe not a use case for AI avatar specifically now but is a use case more generally)

Reply:
I'll add a test for that

Reply:
So we have a few tests for looping such as `can get frames at random times (forward and backward)` and `can get frames when looping`.

Reply:
but I'm guessing those tests don't use the new code path? Like the case for when PTS is greater than what we're looking for